mediorum/ddl/ddl.go

```diff
@@ -76,3 +65,75 @@ func md5string(s string) string {
 	hash := md5.Sum([]byte(s))
 	return hex.EncodeToString(hash[:])
 }
+
+func migrateShardBucket(db *sql.DB, bucket *blob.Bucket) {
+	h := "shardBucket"
+
+	var alreadyRan bool
+	db.QueryRow(`select count(*) = 1 from mediorum_migrations where hash = $1`, h).Scan(&alreadyRan)
+	if alreadyRan {
+		fmt.Printf("hash %s exists skipping ddl \n", h)
+		return
+	}
+
+	ctx := context.Background()
+
+	// collect all keys
+	keys := []string{}
+	iter := bucket.List(nil)
+	for {
+		obj, err := iter.Next(ctx)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatalf("error listing bucket: %v", err)
+		}
+		if strings.HasPrefix(obj.Key, "ba") { // ignore "my_cuckoo" key and keys that already migrated (in case of restart halfway through)
+			keys = append(keys, obj.Key)
+		}
+	}
+
+	var wg sync.WaitGroup
+	sem := make(chan struct{}, 5) // semaphore to limit 5 uploads at a time
```

Comment:
errgroup is nice because you can [SetLimit](https://pkg.go.dev/golang.org/x/sync/errgroup#Group.SetLimit)